### PR TITLE
automake and autoconf improvements

### PR DESCRIPTION
**Pull request by kloczek**
- remove not used `include $(GLIB_MAKEFILE)` lines
- use automake to build intel_lpmd_control
- use autoconf AC_CONFIG_FILES() to generate data/intel_lpmd.service and data/org.freedesktop.intel_lpmd.service instead custom sed rules
- more [] quotes in configure ac
- use $rundir and bumped minimum autoconf version required to 2.70
- remove use AM_CFLAGS in main Makefile.am
- remove . from SUBDIRS in main Makefile.am
- use PKG_CHECK_MODULES() to detect dbus-glib-1
- remove redundant AC_SUBST(FOO_CFLAGS) and AC_SUBST(FOO_LIBS) (PKG_CHECK_MODULES() adds that automatically)
- simplify man pages installation (use automake man_MANS)

### CODE REVIEW

**Comment by kloczek** [comment]
Also `distcheck` target should be working now as well.

File: tests/lpm_test_interface.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/sh
```
Comment (eli-schwartz) [inline tests/lpm_test_interface.sh:1]: > This script is pure POSIX sh so it do not need to be as bash script.

This is the wrong solution to marking a script as POSIX sh. Please use /bin/sh exclusively for this, as that is the portable filename for the "sh" command.

Comment (kloczek) [reply tests/lpm_test_interface.sh:1]: Please notice that currently NONE ot the OS distros (Linux/Solaris/*BSD) uses /usr separation and all those OSess distros uses /usr/bin/sh as standard POSIX sh location (Solaris and *BSD uses that more than decade).
Just look on Fedora or Debian packaging standards.
Fedora rpm currently **automatically** changed /bin/ to /usr/bin/ in all shells shebangs https://src.fedoraproject.org/rpms/redhat-rpm-config/blob/rawhide/f/brp-mangle-shebangs

Comment (eli-schwartz) [reply tests/lpm_test_interface.sh:1]: > Please notice that currently NONE ot the OS distros (Linux/Solaris/*BSD) uses /usr separation and all those OSess distros uses /usr/bin/sh as standard POSIX sh location (Solaris and *BSD uses that more than decade).

Clearly you have no idea what you're talking about in the slightest.

```
$ uname -a
FreeBSD build 14.0-RELEASE-p5 FreeBSD 14.0-RELEASE-p5 #0: Tue Feb 13 23:37:36 UTC 2024     root@amd64-builder.daemonology.net:/usr/obj/usr/src/amd64.amd64/sys/GENERIC amd64

$ file /usr/bin/sh /bin/sh
/usr/bin/sh: cannot open `/usr/bin/sh' (No such file or directory)
/bin/sh:     ELF 64-bit LSB pie executable, x86-64, version 1 (FreeBSD), dynamically linked, interpreter /libexec/ld-elf.so.1, for FreeBSD 14.0 (1400097), FreeBSD-style, stripped
```

Not only is /usr/bin/sh not the "standard POSIX sh location", it isn't a valid location at all. It doesn't exist. It never did and never has and most likely never will.

It's generally unlikely to find /usr/bin/sh except as a side effect of Linux-specific systems that happen to have implemented /bin as a symlink to /usr/bin, which isn't even universal on Linux either. You won't get very far on Alpine Linux, for example, if you assume that /usr/bin/sh exists.

This is why it is NOT the "standard POSIX sh location".

Comment (zhang-rui) [reply tests/lpm_test_interface.sh:1]: Hi, guys
May I know if we are on the same page now, for the discussion above?

**Comment by kloczek** [comment]
Thx 👍

**Comment by zhang-rui** [comment]
hi @kloczek 
Just noticed that only the first patch has your SOB.
Can you please update the PR with SOB in each commit?

**Comment by kloczek** [comment]
SOB?

**Comment by zhang-rui** [comment]
"Signed-off-by" tag :)

**Comment by kloczek** [comment]
It is "Signed-off-by: Tomasz Kłoczko <kloczek@github.com>"

**Pull request merged**